app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -1,148 +1,166 @@
+  }
 
-    self.hasScrolled = function () {
-      _hasScrolled = true
-      return _hasScrolled
-    }
+  HighlightActiveSectionHeading.prototype.hasScrolled = function () {
+    this._hasScrolled = true
+    return this._hasScrolled
+  }
 
-    self.checkResize = function () {
-      if (_hasResized) {
-        _hasResized = false
-        _hasScrolled = true
-      }
+  HighlightActiveSectionHeading.prototype.checkResize = function () {
+    if (this._hasResized) {
+      this._hasResized = false
+      this._hasScrolled = true
     }
+  }
 
-    self.checkScroll = function () {
-      if (_hasScrolled) {
-        _hasScrolled = false
-        var windowDimensions = self.getWindowDimensions()
-        if (windowDimensions.width <= 768) {
-          self.removeActiveItem()
-        } else {
-          self.updateActiveNavItem()
-        }
+  HighlightActiveSectionHeading.prototype.checkScroll = function () {
+    if (this._hasScrolled) {
+      this._hasScrolled = false
+      var windowDimensions = this.getWindowDimensions()
+      if (windowDimensions.width <= 768) {
+        this.removeActiveItem()
+      } else {
+        this.updateActiveNavItem()
       }
     }
+  }
 
-    self.getAnchors = function () {
-      $.each(self.$anchors, function (i) {
-        var anchorID = $(this).attr('href')
-        // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
-        anchorIDs.push(anchorID)
-      })
+  HighlightActiveSectionHeading.prototype.getWindowDimensions = function () {
+    return {
+      height: window.innerHeight,
+      width: window.innerWidth
     }
+  }
 
-    self.getHeadingPosition = function ($theID) {
-      return $theID.offset()
+  HighlightActiveSectionHeading.prototype.getAnchors = function () {
+    for (var i = 0; i < this.anchors.length; i++) {
+      var anchorID = this.anchors[i].getAttribute('href')
+      // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
+      this.anchorIDs.push(anchorID)
     }
+  }
 
-    self.getNextHeadingPosition = function ($theNextID) {
-      return $theNextID.offset()
+  HighlightActiveSectionHeading.prototype.updateActiveNavItem = function () {
+    var windowVerticalPosition = this.getWindowPositions()
+    var footer = document.querySelector('.govuk-footer')
+    if (footer) {
+      var footerPosition = this.getFooterPosition(footer)
     }
 
-    self.getFooterPosition = function ($theID) {
-      return $theID.offset()
-    }
+    for (var i = 0; i < this.anchors.length; i++) {
+      var theID = this.anchorIDs[i]
+      var theNextID = this.anchorIDs[i + 1]
 
-    self.getDistanceBetweenHeadings = function (headingPosition, nextHeadingPosition) {
-      var distanceBetweenHeadings = (nextHeadingPosition - headingPosition)
-      return distanceBetweenHeadings
-    }
+      var $theID = document.getElementById(theID.substring(1)) // remove the # at the start
```

Comment:
Do we ever need the anchor id without the "#" ? should this substring(1) be done in `getAnchors` ?

Reply:
Ideally yes, but we use it with the `#` in `setActiveItem`, where we compare the value of the link href attribute `#example` with the anchor ID, which if we'd already removed the `#` would be `example`.

So either we strip it out when we collect them, then add them back in again when highlighting the links, or we keep it when we collect them, and strip it out when finding the elements. It's a bit confusing either way.

Reply:
Ah fair - sorry!